File: server/bcephal-grid/src/main/java/com/moriset/bcephal/grid/domain/JoinLogBrowserData.java
/**
 * 
 */
package com.moriset.bcephal.grid.domain;

import java.sql.Timestamp;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.domain.DataSourceType;
import com.moriset.bcephal.domain.RunModes;
import com.moriset.bcephal.domain.RunStatus;
import com.moriset.bcephal.utils.JsonDateTimeDeserializer;
import com.moriset.bcephal.utils.JsonDateTimeSerializer;

import jakarta.persistence.Column;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author Joseph Wambo
 *
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class JoinLogBrowserData extends BrowserData {

	private Long joinId;
	
	@Enumerated(EnumType.STRING) 
	private JoinPublicationMethod publicationMethod;
	
	@Enumerated(EnumType.STRING)
	private DataSourceType publicationGridType;
	
	private String publicationGridName;
	
	private String publicationNumber;
	
	private Long publicationNbrAttributeId;	
	
	private String publicationNbrAttributeName;
	
	@JsonSerialize(using = JsonDateTimeSerializer.class)
	@JsonDeserialize(using = JsonDateTimeDeserializer.class)
	@Column(nullable = false)
	private Timestamp endDate;
	
	@Enumerated(EnumType.STRING) 
	private RunStatus status;
	
	@Enumerated(EnumType.STRING) 
	private RunModes mode;
		
	private String user;
		
	private long rowCount;
	
	private String message;
	
	
	public JoinLogBrowserData(JoinLog log) {
		super(log.getId(), log.getName(), log.getCreationDate(), log.getModificationDate());		
		this.joinId = log.getJoinId();	
		publicationMethod = log.getPublicationMethod();
		publicationGridType = log.getPublicationGridType();
		publicationGridName = log.getPublicationGridName();		
		publicationNumber = log.getPublicationNumber();		
		endDate = log.getEndDate();		
		status = log.getStatus();		
		mode = log.getMode();			
		user = log.getUser();			
		publicationNbrAttributeId = log.getPublicationNbrAttributeId();		
		publicationNbrAttributeName = log.getPublicationNbrAttributeName();		
		rowCount = log.getRowCount();
		message = log.getMessage();		
	}
	
}

File: client/bcephal-client-model/Joins/JoinLogBrowserData.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;

namespace Bcephal.Models.Joins
{
    public class JoinLogBrowserData : BrowserData
    {
		public long? JoinId { get; set; }

		public string PublicationGridName { get; set; }

		public string PublicationNumber { get; set; }

		public long? PublicationNbrAttributeId { get; set; }

		public string PublicationNbrAttributeName { get; set; }

		public string EndDate { get; set; }

		public string Status { get; set; }

		public string Mode { get; set; }

		public string User { get; set; }

		public long? RowCount { get; set; }

		public string Message { get; set; }

		[JsonIgnore]
		public RunStatus RunStatus
		{
			get { return !string.IsNullOrEmpty(Status) ? RunStatus.GetByCode(Status) : null; }
			set { this.Status = value != null ? value.code : null; }
		}

		[JsonIgnore]
		public RunModes RunMode
		{
			get { return !string.IsNullOrEmpty(Mode) ? RunModes.GetByCode(Mode) : null; }
			set { this.Mode = value != null ? value.code : null; }
		}

		[JsonIgnore]
		public DateTime? EndDateTime
		{
			get
			{
				try
				{
					return DateUtils.ParseDateTime(EndDate);
				}
				catch (Exception)
				{
					return null;
				}
			}
		}
	}
}
